### PR TITLE
feat: support date picker in action form widgets

### PR DESCRIPTION
**Pull request by realSpok**
## Definition of Done
#85yxm8nvq
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-customizer/src/decorators/actions/types/fields.ts [hunk 1]
```diff
@@ -78,6 +78,14 @@ type TextInputFieldConfiguration = {
   placeholder?: string;
 };
 
+type DateInputFieldConfiguration<Context> = {
+  widget: 'Date';
```
Comment (ghusse) [inline packages/datasource-customizer/src/decorators/actions/types/fields.ts:82]: I think the name should reflect that the widget is asking for a date + time here

File: packages/datasource-customizer/src/decorators/actions/types/fields.ts [hunk 2]
```diff
@@ -78,6 +78,14 @@ type TextInputFieldConfiguration = {
...
+  format?: ValueOrHandler<Context, string>;
+  placeholder?: string;
+  minDate?: ValueOrHandler<Context, Date>;
```
Comment (ghusse) [inline packages/datasource-customizer/src/decorators/actions/types/fields.ts:85]: why not simply `min` and `max`?

**Review by ghusse — CHANGES_REQUESTED** [review]
- ⚠️&nbsp;Code review
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ✖️&nbsp; Integration tests
- ⚠️&nbsp;Manual tests
- ✅&nbsp;PR title
- ✅&nbsp;Destination branch
- ✅&nbsp;PR linked to the clickup task


I suggest renaming options `min` and `max`, and rename the Widget to `DateTimePicker` or `DateTime` to indicate that a time is also requested.

In addition to that, the current `minDate` property does not seem to work. I tested to provide a value but was able to select a date before this date in my form.

File: packages/datasource-customizer/src/decorators/actions/types/fields.ts (hunk 2 above)
Comment (realSpok) [reply packages/datasource-customizer/src/decorators/actions/types/fields.ts:85]: I used the same name as in the date component on frontend side, for consistency

File: packages/agent/src/utils/forest-schema/action-fields.ts [hunk 3]
```diff
@@ -73,6 +74,12 @@ export default class ActionFields {
     return (field as ActionFieldTextInput)?.widget === 'TextInput';
   }
 
+  public static isDateInputField(
```
Comment (ghusse) [inline packages/agent/src/utils/forest-schema/action-fields.ts:77]:
```suggestion
  public static isDateTimeField(
```

File: packages/datasource-customizer/src/decorators/actions/types/fields.ts [hunk 4]
```diff
@@ -78,6 +78,14 @@ type TextInputFieldConfiguration = {
   placeholder?: string;
 };
 
+type DateInputFieldConfiguration<Context> = {
```
Comment (ghusse) [inline packages/datasource-customizer/src/decorators/actions/types/fields.ts:81]:
```suggestion
type DateTimeFieldConfiguration<Context> = {
```

File: packages/datasource-toolkit/src/interfaces/action.ts [hunk 5]
```diff
@@ -92,6 +92,15 @@ export type ActionFieldTextInput = ActionFieldBase & {
   placeholder?: string;
 };
 
+export type ActionFieldDateInput = ActionFieldBase & {
```
Comment (ghusse) [inline packages/datasource-toolkit/src/interfaces/action.ts:95]:
```suggestion
export type ActionFieldDateTime = ActionFieldBase & {
```

File: packages/forestadmin-client/src/schema/types.ts [hunk 6]
```diff
@@ -119,6 +119,16 @@ export type ForestServerActionFieldTextInputOptions = {
   };
 };
 
+export type ForestServerActionFieldDateInputOptions = {
```
Comment (ghusse) [inline packages/forestadmin-client/src/schema/types.ts:122]:
```suggestion
export type ForestServerActionFieldDateTimeOptions = {
```

File: packages/datasource-toolkit/src/interfaces/action.ts [hunk 7]
```diff
@@ -92,6 +92,15 @@ export type ActionFieldTextInput = ActionFieldBase & {
   placeholder?: string;
 };
 
+export type ActionFieldDatePickerInput = ActionFieldBase & {
```
Comment (ghusse) [inline packages/datasource-toolkit/src/interfaces/action.ts:95]: A declaration is missing on the type `ActionFieldWidget` below

File: packages/agent/src/utils/forest-schema/generator-action-field-widget.ts [hunk 8]
```diff
@@ -144,6 +148,20 @@ export default class GeneratorActionFieldWidget {
...
 
+  private static buildDatePickerInputWidgetEdit(
+    field: ActionField,
+  ): ForestServerActionFieldDatePickerInputOptions {
+    return {
+      name: 'date editor',
+      parameters: {
+        format: field.format || null,
+        placeholder: field.placeholder || null,
+        minDate: field.min instanceof Date ? field.min.toDateString() : null,
```
Comment (ghusse) [inline packages/agent/src/utils/forest-schema/generator-action-field-widget.ts:159]:
```suggestion
        minDate: field.min instanceof Date ? field.min.toISOString() : null,
```

File: packages/agent/test/utils/forest-schema/generator-action-field-widget.test.ts [hunk 9]
```diff
@@ -733,6 +733,81 @@ describe('GeneratorActionFieldWidget', () => {
...
+            label: 'Label',
+            watchChanges: false,
+            widget: 'DatePicker',
+            [parameter]: date,
+          });
+
+          expect(result).toMatchObject({
+            name: 'date editor',
+            parameters: {
+              [`${parameter}Date`]: date.toDateString(),
```
Comment (ghusse) [inline packages/agent/test/utils/forest-schema/generator-action-field-widget.test.ts:788]: It's better here to hardcode values, to repeating the same error twice: in the code and in tests.

**Review by ghusse — APPROVED** [review]
- ✅&nbsp;Code review
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ✖️&nbsp; Integration tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ✅&nbsp;Destination branch
- ✅&nbsp;PR linked to the clickup task

**Pull request merged**